system/jlib/jstats.cpp

```diff
@@ -1994,12 +2098,82 @@ class CStatisticCollection : public CInterfaceOf<IStatisticCollection>
             cur.visit(visitor);
     }
 
+    virtual bool refreshAggregates(CRuntimeStatisticCollection & totals, IBitSet & isTotalUpdated) override
+    {
+        const StatisticsMapping & mapping = totals.queryMapping();
+        bool updated = false;
+        if (queryScopeType()==SSTsubgraph || isDirty==false)
```

Comment:
I agree: checking isDirty is sufficient.  I had the subgraph test because the isDirty flag was implemented later and I didn't reassess the need for the subgraph test.

I modified the code so that if isDirty is false at the global level, the operation is a nop.

Reply:
Ignore the previous comment.  The unconditional sg test is required.  The first block (if condition true) returns the totals at this level.  The second block (if condition false) descends to a lower scope level to get totals from lower level.

It should return totals from the current level under 2 circumstances
1) isDirty==false: the stats at this level and descendant levels haven't changed, so it is ok to use the aggregates from this level
2) it is at subgraph scope:  this condition is required because even if the dirty flag is true, aggregates should only be generated from subgraph level.  As per the agreed requirements, the aggregates generated automatically should not be generated from activities->subgraph.  (At a later date, if it is determined that aggregates should be generated from aggregates, this condition may be removed but it is beyond the scope of this jira for this change now.)